src/Translatable/Translatable.php

```diff
@@ -607,4 +612,42 @@ protected function locale()
         return App::make('config')->get('translatable.locale')
             ?: App::make('translator')->getLocale();
     }
+
+    /**
+     * Inner join with the translation table
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param null $locale
+     * @return this
+     */
+    public static function scopeJoinTranslation(Builder $query, $locale = null)
+    {
+        // models
+        $instance = new static;
+
+        $translationModelName = $instance->getTranslationModelName();
+        $translationModel = new $translationModelName();
+
+        // locale
+        if (is_null($locale)) {
+            $locale = $instance->locale();
+        }
+
+        return $query->join($translationModel->getTable() . ' as t', 't.' . $instance->getRelationKey(), '=', $instance->getTable() . '.' . $instance->getKeyName())
+            ->where($instance->getLocaleKey(), $locale);
+    }
+    
+    /**
+     * Check if translation was with a inner join
+     *
+     * @return bool
+     */
+    protected function isJoinTranslated(){
+        foreach($this->translatedAttributes as $translatedAttribute){
+            if(isset($this->original[$translatedAttribute]) && !empty($this->original[$translatedAttribute])){
```

Comment:
There is a case when $this->original[$translatedAttribute] is empty string - this doesn't mean it is not translated. In these cases an additional query is issued to the database.